HeterogeneousCore/CUDAUtilities/test/radixSort_t.cu

```diff
@@ -32,6 +34,50 @@ struct RS<float> {
   static constexpr int imax = std::numeric_limits<int>::max();
 };
 
+// A templated unsigned integer type with N bytes
+template <int N>
+struct uintN;
+
+template <>
+struct uintN<8> {
+  using type = uint8_t;
+};
+
+template <>
+struct uintN<16> {
+  using type = uint16_t;
+};
+
+template <>
+struct uintN<32> {
+  using type = uint32_t;
+};
+
+template <>
+struct uintN<64> {
+  using type = uint64_t;
+};
+
+template <int N>
+using uintN_t = typename uintN<N>::type;
+
+// A templated unsigned integer type with the same size as T
+template <typename T>
+using uintT_t = uintN_t<sizeof(T) * 8>;
+
+// Keep only the `N` most significant bytes of `t`, and set the others to zero
+template <int N, typename T, typename SFINAE = std::enable_if_t<N <= sizeof(T)>>
+void truncate(T& t) {
+  const int shift = 8 * (sizeof(T) - N);
+  union {
+    T t;
+    uintT_t<T> u;
+  } c;
+  c.t = t;
+  c.u = c.u >> shift << shift;
```

Comment:
I'm pretty sure this is actually undefined behavior as you are not supposed to set a union using one type and read the union using another type. I believe the recommend way to do this is to use `memcpy`.
```C++
uintT_t<T> u;
memcpy(&u, &t, sizeof(u));
u = u >>shift <<shift;
memcpy(&t, &u, sizeof(t));
```

I played around with that recently on godbolt (for reasons other than this PR) and found compilers know about memcpy and optimize out the calls and just do the 'right' thing.

Reply:
I think section 11.5.6.3 of the C++ 20 standard demonstrates that it is undefined behavior.

Reply:
I'm pretty sure the union pattern is used elsewhere in CMSSW too, so we should address those as well.